packages/accounts-base/localstorage_token.js

```diff
@@ -25,116 +32,140 @@ Accounts._enableAutoLogin = function () {
+  Meteor._localStorage.removeItem(this.LOGIN_TOKEN_KEY);
+  Meteor._localStorage.removeItem(this.LOGIN_TOKEN_EXPIRES_KEY);
 
   // to ensure that the localstorage poller doesn't end up trying to
   // connect a second time
-  lastLoginTokenWhenPolled = null;
+  this._lastLoginTokenWhenPolled = null;
 };
 
 // This is private, but it is exported for now because it is used by a
 // test in accounts-password.
 //
-storedLoginToken = Accounts._storedLoginToken = function() {
-  return Meteor._localStorage.getItem(loginTokenKey);
+Ap._storedLoginToken = function () {
+  return Meteor._localStorage.getItem(this.LOGIN_TOKEN_KEY);
 };
 
-storedLoginTokenExpires = function () {
-  return Meteor._localStorage.getItem(loginTokenExpiresKey);
+Ap._storedLoginTokenExpires = function () {
+  return Meteor._localStorage.getItem(this.LOGIN_TOKEN_EXPIRES_KEY);
 };
 
-var storedUserId = function() {
-  return Meteor._localStorage.getItem(userIdKey);
+Ap._storedUserId = function () {
+  return Meteor._localStorage.getItem(this.USER_ID_KEY);
 };
 
-var unstoreLoginTokenIfExpiresSoon = function () {
-  var tokenExpires = Meteor._localStorage.getItem(loginTokenExpiresKey);
-  if (tokenExpires && Accounts._tokenExpiresSoon(new Date(tokenExpires)))
-    unstoreLoginToken();
+Ap._unstoreLoginTokenIfExpiresSoon = function () {
+  var tokenExpires = this._storedLoginTokenExpires();
+  if (tokenExpires && this._tokenExpiresSoon(new Date(tokenExpires))) {
+    self._unstoreLoginToken();
+  }
 };
 
 ///
 /// AUTO-LOGIN
 ///
 
-if (autoLoginEnabled) {
-  // Immediately try to log in via local storage, so that any DDP
-  // messages are sent after we have established our user account
-  unstoreLoginTokenIfExpiresSoon();
-  var token = storedLoginToken();
-  if (token) {
-    // On startup, optimistically present us as logged in while the
-    // request is in flight. This reduces page flicker on startup.
-    var userId = storedUserId();
-    userId && Accounts.connection.setUserId(userId);
-    Meteor.loginWithToken(token, function (err) {
-      if (err) {
-        Meteor._debug("Error logging in with token: " + err);
-        makeClientLoggedOut();
-      }
-      Accounts._pageLoadLogin({
-        type: "resume",
-        allowed: !err,
-        error: err,
-        methodName: "login",
-        // XXX This is duplicate code with loginWithToken, but
-        // loginWithToken can also be called at other times besides
-        // page load.
-        methodArguments: [{resume: token}]
+Ap._initLocalStorage = function () {
+  var self = this;
+
+  // Key names to use in localStorage
+  self.LOGIN_TOKEN_KEY = "Meteor.loginToken";
+  self.LOGIN_TOKEN_EXPIRES_KEY = "Meteor.loginTokenExpires";
+  self.USER_ID_KEY = "Meteor.userId";
+
+  if (this.connection !== Meteor.connection) {
+    // If this.connection is not simply Meteor.connection, append a
+    // deterministic connection-specific suffix to the key used with
+    // Meteor._localStorage.
```

Comment:
Using different keys with `Meteor._localStorage` depending on the URL of the DDP connection is really important! So is using the same keys as before when `self.connection === Meteor.connection` (so that folks won't have to log in again when we release these changes).